main_agent.py

```diff
@@ -0,0 +1,97 @@
+import os
+from typing import Optional
+
+import streamlit as st
+from dotenv import load_dotenv
+from langchain.agents import AgentType, initialize_agent, load_tools
+from langchain.callbacks import StreamlitCallbackHandler
+from langchain.callbacks.manager import AsyncCallbackManagerForToolRun, CallbackManagerForToolRun
+from langchain.chains import ConversationalRetrievalChain
+from langchain.chat_models import ChatOpenAI
+from langchain.memory import ConversationBufferMemory, StreamlitChatMessageHistory
+from langchain.tools import BaseTool
+
+from src.taix.client import wv_retriever, wv_retriever_limits
+
+load_dotenv()
+
+st.set_page_config(page_title="TAix", page_icon="💸")
+st.title("💸 TAix - Tax Advice Agent")
+
+
+class TaxiLimitsTool(BaseTool):
+    name = "tax_limit_tool"
+    description = "useful for when you need to get information tax limits"
+
+    def _run(self, query: str, run_manager: Optional[CallbackManagerForToolRun] = None) -> str:
+        """Use the tool."""
+
+        msgs = StreamlitChatMessageHistory()
+        memory = ConversationBufferMemory(memory_key="chat_history", chat_memory=msgs, return_messages=True)
+        w_url = os.getenv("WEAVIATE_URL", "http://localhost:8080")
+        vector_store = wv_retriever_limits(w_url)
+
+        # Setup LLM and QA chain
+        llm = ChatOpenAI(model_name="gpt-4", temperature=0, streaming=True)
+        qa_chain = ConversationalRetrievalChain.from_llm(llm, retriever=vector_store, verbose=True, memory=memory)
+        query_template = f"What is the tax limit? Addinoal information: {query}"
```

Comment:
Change to f"What is the tax limit? Additional information: {query}"
Typo fix, to not confuse gpt